core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -425,8 +428,28 @@ private void tryStart(Instant startedAt) {
             // For all registered output consumers, start following as close to container startup as possible
             this.logConsumers.forEach(this::followOutput);
 
+            // Wait until inspect container returns the mapped ports
+            containerInfo = await()
+                .atMost(5, TimeUnit.SECONDS)
+                .pollInterval(DynamicPollInterval.ofMillis(50))
+                .pollInSameThread()
+                .until(
+                    () -> dockerClient.inspectContainerCmd(containerId).exec(),
+                    inspectContainerResponse -> {
+                         Set<Integer> currentlyExposedPorts = inspectContainerResponse
+                            .getNetworkSettings()
+                            .getPorts()
+                            .getBindings()
+                            .keySet()
+                            .stream()
+                            .map(ExposedPort::getPort)
```

Comment:
I think we need to check whether it is actually mapped (the original issue) vs just present in the response but the mapping may still be pending

Reply:
You are absolutely right.